app/v2/models/MovementType.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.models
+
+import play.api.libs.json.JsError
+import play.api.libs.json.JsString
+import play.api.libs.json.JsSuccess
+import play.api.libs.json.Reads
+import play.api.libs.json.Writes
+
+sealed abstract class MovementType(val movementType: String)
+
+object MovementType {
+  case object Departure extends MovementType("departure")
+  case object Arrival   extends MovementType("arrival")
+
+  implicit val movementTypeReads: Reads[MovementType] = Reads {
+    case x: JsString => MovementType.findByName(x.value).map(JsSuccess(_)).getOrElse(JsError())
```

Comment:
Would writing it something like this read better?

`case JsString(movementType) if MovementType.findByName(movementType) => JsSuccess(movementType)
case _ => JsError()`

Reply:
Oh yes, my bad.